ir/src/symbol_table.rs

```diff
@@ -20,16 +21,48 @@ impl Display for IdentifierType {
         match self {
             Self::PublicInput(_) => write!(f, "PublicInput"),
             Self::PeriodicColumn(_, _) => write!(f, "PeriodicColumn"),
-            Self::MainTraceColumn(_) => write!(f, "MainTraceColumn"),
-            Self::AuxTraceColumn(_) => write!(f, "AuxTraceColumn"),
+            Self::TraceColumn(column) => {
+                write!(f, "TraceColumn in segment {}", column.trace_segment())
+            }
         }
     }
 }
 
+/// Describes a column in the execution trace by the trace segment to which it belongs and its
+/// index within that segment.
+#[derive(Debug, Copy, Clone)]
+pub struct TraceColumn {
+    trace_segment: TraceSegment,
+    col_idx: usize,
+}
+
+impl TraceColumn {
+    /// Creates a [TraceColumn] in the specified trace segment at the specified index.
+    fn new(trace_segment: TraceSegment, col_idx: usize) -> Self {
+        Self {
+            trace_segment,
+            col_idx,
+        }
+    }
+
+    /// Gets the trace segment of this [TraceColumn].
+    pub fn trace_segment(&self) -> TraceSegment {
+        self.trace_segment
+    }
+
+    /// Gets the column index of this [TraceColumn].
+    pub fn col_idx(&self) -> usize {
+        self.col_idx
+    }
+}
+
 /// SymbolTable for identifiers to track their types and information and enforce uniqueness of
 /// identifiers.
 #[derive(Default, Debug)]
 pub(super) struct SymbolTable {
+    /// The number of trace segments in the AIR.
+    num_trace_segments: TraceSegment,
```

Comment:
nit: I found it a bit odd that the type of `num_trace_segments` is `TraceSegment`. I know it's just an alias for `u8`, but it looks strange because I think of `TraceSegment` as in index of trace segment. Maybe this should be just `u8` or `usize`?